src/SchemaDesign/SchemaDesigner.tsx

```diff
@@ -0,0 +1,418 @@
+/* eslint-disable react/jsx-props-no-spreading */
```

Comment:
I'm not a fan of disabling this rule. The docs give some good reasons as to why it's useful to follow:

"Enforces that there is no spreading for any JSX attribute. This enhances readability of code by being more explicit about what props are received by the component. It is also good for maintainability by avoiding passing unintentional extra props and allowing react to emit warnings when invalid HTML props are passed to HTML elements."

Can we instead destructure the object and explicitly provide the props we need?

Reply:
I agree, I'm happy to move it to be global. Some rules like this can be really nit-picky when sometimes there's no issue spreading props.